sklearn/feature_selection/mutual_info.py

```diff
@@ -0,0 +1,293 @@
+        data will be overwritten if it was in dense format.
+
+    Attributes
+    ----------
+    support_ : ndarray, shape (n_features,)
+        Mask of selected features.
+
+    relevance_ : ndarray, shape (n_features,)
+        Array containing mutual information values between each feature and
+        the target variable.
+
+    redundancy_ : ndarray, shape (n_features, n_features)
+        Array containing mutual information values between each pair of
+        features. Note, that diagonal elements are not computed and contain
+        some random numbers.
+
+    References
+    ----------
+    .. [1] H. Peng, F. Long, and C. Ding, "Feature selection based on mutual
+           information: criteria of max-dependency, max-relevance, and
+           min-redundancy", IEEE Transactions on Pattern Analysis and Machine
+           Intelligence, Vol. 27, No. 8, pp. 1226-1238, 2005.
+    .. [2] A. Kraskov, H. Stogbauer and P. Grassberger, "Estimating mutual
+           information". Phys. Rev. E 69, 2004.
+    .. [3] B. C. Ross "Mutual Information between Discrete and Continuous
+           Data Sets". PLoS ONE 9(2), 2014.
+    """
+    def __init__(self, n_features_to_select=0.5, categorical_features=False,
+                 categorical_target=False, n_neighbors=3, copy=True):
+        self.n_features_to_select = n_features_to_select
+        self.categorical_features = categorical_features
+        self.categorical_target = categorical_target
+        self.n_neighbors = n_neighbors
+        self.copy = copy
+
+    def _get_support_mask(self):
+        return self.support_
+
+    def fit(self, X, y):
+        """Find the best features according to mutual information criterion.
+
+        Parameters
+        ----------
+        X : array_like or sparse matrix, shape (n_samples, n_features)
+            Feature matrix.
+        y : array_like, shape (n_samples,)
+            Target vector.
+
+        Returns
+        -------
+        self
+        """
+        X, y = check_X_y(X, y, accept_sparse='csr',
+                         y_numeric=not self.categorical_target)
+        n_features = X.shape[1]
+
+        if self.n_features_to_select >= 1:
+            n_features_to_select = int(self.n_features_to_select)
+        elif 0 < self.n_features_to_select < 1:
+            n_features_to_select = max(
+                1, int(self.n_features_to_select * n_features))
+        else:
+            raise ValueError("`n_features_to_select` must be positive.")
+
+        if isinstance(self.categorical_features, bool):
+            categorical_features = np.empty(n_features, dtype=bool)
+            categorical_features.fill(self.categorical_features)
+        else:
+            categorical_features = np.asarray(self.categorical_features)
+
+        # Scale all variables, regardless of being categorical or not.
+        # It's simple, clear and convenient.
+        X = scale(X, with_mean=False, copy=self.copy)
```

Comment:
"Although this is not required,  strictly speaking it will  reduce  errors  in  general".

Do you know what type of errors, they are talking about, is it numerical?